app/portainer/views/settings/authentication/settingsAuthenticationController.js

```diff
@@ -69,6 +77,13 @@ function($q, $scope, $state, Notifications, SettingsService, FileUploadService,
 
   $scope.saveSettings = function() {
     var settings = $scope.settings;
+
+    // Delete credentials if in Anonymous Mode
+    if ($scope.formValues.Anonymous){
```

Comment:
I got thinking, if an empty-string `""` will be considered as an ordinary settable value in the API, then we must make sure to unset the Password attribute also for authenticated mode (non-anonymous) if the field is left empty, i.e. the user is not changing the current password.